runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/executor/PythonExecutorCommunicationServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+
+@Service("pythonExecutorCommunicationServiceImpl")
+public class PythonExecutorCommunicationServiceImpl implements PythonExecutorCommunicationService {
+
+    private static final String EXTERNAL_PYTHON_EXECUTOR_STOP_PATH = "/rest/v1/stop";
+    private static final String EXTERNAL_PYTHON_EXECUTOR_HEALTH_PATH = "/rest/v1/health";
+    private static final String EXTERNAL_PYTHON_EXECUTOR_EVAL_PATH = "/rest/v1/eval";
+    private final ResteasyClient restEasyClient;
+
+    private final PythonExecutorConfigurationDataService pythonExecutorConfigurationDataService;
+
+    @Autowired
+    public PythonExecutorCommunicationServiceImpl(StatefulRestEasyClientsHolder statefulRestEasyClientsHolder,
+                                                  @Qualifier("pythonExecutorConfigurationDataService") PythonExecutorConfigurationDataService pythonExecutorConfigurationDataService) {
+        this.restEasyClient = statefulRestEasyClientsHolder.getRestEasyClient();
+        this.pythonExecutorConfigurationDataService = pythonExecutorConfigurationDataService;
+    }
+
+    @Override
+    public boolean isAlivePythonExecutor() {
+        try (Response response = restEasyClient
+                .target(pythonExecutorConfigurationDataService.getPythonExecutorConfiguration().getUrl())
+                .path(EXTERNAL_PYTHON_EXECUTOR_HEALTH_PATH)
+                .request()
+                .accept(APPLICATION_JSON_TYPE)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .build("GET")
+                .invoke()) {
+            return response.getStatus() == 200;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    @Override
+    public Response stopPythonExecutor() {
+        PythonExecutorDetails pythonExecutorConfiguration = pythonExecutorConfigurationDataService.getPythonExecutorConfiguration();
+        return restEasyClient
+                .target(pythonExecutorConfiguration.getUrl())
+                .path(EXTERNAL_PYTHON_EXECUTOR_STOP_PATH)
+                .request()
+                .accept(APPLICATION_JSON_TYPE)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .header(AUTHORIZATION, pythonExecutorConfiguration.getLifecycleEncodedAuth())
+                .build("POST")
+                .invoke();
+    }
+
+    @Override
+    public Response executeRequestOnPythonServer(String method, String payload) {
+        return restEasyClient
+                .target(getPythonExecutorConfiguration().getUrl())
+                .path(EXTERNAL_PYTHON_EXECUTOR_EVAL_PATH)
+                .request()
+                .accept(APPLICATION_JSON_TYPE)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .header(AUTHORIZATION, getPythonExecutorConfiguration().getRuntimeEncodedAuth())
+                .build(method, entity(payload, APPLICATION_JSON_TYPE))
+                .invoke();
+    }
+
+    @Override
+    public PythonExecutorDetails getPythonExecutorConfiguration() {
```

Comment:
This shouldn't exist here, one may use pythonExecutorConfigurationDataService from other classes too.